network/src/interface/mod.rs

```diff
@@ -105,6 +107,7 @@ where
         );
         let peer_handle = PeerHandle::new(peer_id, peer_reqs_tx);
         let peer = Peer::new(
+            Arc::clone(&network_context),
```

Comment:
What's the difference between `Arc::clone(obj)` and `obj.clone()`

Reply:
They're the exact same behaviour. At some point we had a coding guideline to use `Arc::clone(&my_arc)` instead of `my_arc.clone()` so we could easily grep all the `Arc::clone`'s we were doing (for perf debugging or something). Though, I can't seem to find this written down in the repo anywhere...

@bmwill: do you remember this or am I just hallucinating? is this still a thing?

Reply:
We actually ripped that out of the coding guidelines as of last week because everyone hated typing the extra characters :)